template-transformer/handler.lua

```diff
@@ -134,30 +141,46 @@ function TemplateTransformerHandler:body_filter(config)
       -- body is fully read
       local headers = res_get_headers()
       local raw_body = ngx.ctx.buffer
-      local body = read_json_body(raw_body)
-      if body == nil then
-        return ngx.ERROR
+      local body = nil
+
+      local content_type = headers['Content-Type']
+      if content_type == nil then
+        content_type = "application/json"
       end
-      local transformed_body = template_transformer.get_template(config.response_template) {headers = headers,
-                                                                                            body = body,
-                                                                                            raw_body = raw_body,
-                                                                                            cjson_encode = cjson_encode,
-                                                                                            cjson_decode = cjson_decode,
-                                                                                            mask_field = utils.mask_field,
-                                                                                            status = ngx.status}
-      local transformed_body_json = prepare_body(transformed_body);
-
-      ngx.arg[1] = transformed_body_json
-
-      if transformed_body_json == nil or transformed_body_json == '' then 
+
+      if gmatch(content_type, "(application/json)")() then
+        body = read_json_body(raw_body)
+        if body == nil then
+          return ngx.ERROR
+        end
+      end
+      
+      local transformed_body = template_transformer.get_template(config.response_template){headers = headers,
```

Comment:
A ideia é só existir `body` se for JSON mesmo ?

Reply:
Foi o que me veio em mente, por hora não vamos "tratar" outros tipos de formato como é feito com json, só trazer do upstream mesmo e passar pra frente.